--- a/net/mptcp/pm_netlink.c	2025-06-27 12:11:46.000000000 +0200
+++ b/net/mptcp/pm_netlink.c	2025-07-15 09:02:05.542343123 +0200
@@ -1902,7 +1902,7 @@
 	return mptcp_pm_dump_addr(msg, cb);
 }
 
-static int parse_limit(struct genl_info *info, int id, unsigned int *limit)
+static int parse_limit(struct genl_info *info, int id, unsigned int *limit, unsigned int max)
 {
 	struct nlattr *attr = info->attrs[id];
 
@@ -1910,7 +1910,7 @@
 		return 0;
 
 	*limit = nla_get_u32(attr);
-	if (*limit > MPTCP_PM_ADDR_MAX) {
+	if (*limit > max) {
 		GENL_SET_ERR_MSG(info, "limit greater than maximum");
 		return -EINVAL;
 	}
@@ -1925,12 +1925,12 @@
 
 	spin_lock_bh(&pernet->lock);
 	rcv_addrs = pernet->add_addr_accept_max;
-	ret = parse_limit(info, MPTCP_PM_ATTR_RCV_ADD_ADDRS, &rcv_addrs);
+	ret = parse_limit(info, MPTCP_PM_ATTR_RCV_ADD_ADDRS, &rcv_addrs, MPTCP_PM_ADDR_MAX);
 	if (ret)
 		goto unlock;
 
 	subflows = pernet->subflows_max;
-	ret = parse_limit(info, MPTCP_PM_ATTR_SUBFLOWS, &subflows);
+	ret = parse_limit(info, MPTCP_PM_ATTR_SUBFLOWS, &subflows, 64);
 	if (ret)
 		goto unlock;
 
